test/libsolidity/StandardCompiler.cpp

```diff
@@ -154,6 +154,42 @@ BOOST_AUTO_TEST_CASE(no_sources)
 	BOOST_CHECK(containsError(result, "JSONError", "No input sources specified."));
 }
 
+BOOST_AUTO_TEST_CASE(no_sources_empty_object)
+{
+	char const* input = R"(
+	{
+		"language": "Solidity",
+		"sources": {}
+	}
+	)";
+	Json::Value result = compile(input);
+	BOOST_CHECK(containsError(result, "JSONError", "No input sources specified."));
+}
+
+BOOST_AUTO_TEST_CASE(no_sources_empty_array)
+{
+	char const* input = R"(
+	{
+		"language": "Solidity",
+		"sources": []
+	}
+	)";
+	Json::Value result = compile(input);
+	BOOST_CHECK(containsError(result, "JSONError", "Sources input is not a JSON object."));
```

Comment:
How about `Field \"sources\" is not a JSON object` (or maybe omit `field`)?

Reply:
Yeah, thats better. With quotes it's  much more clear. I would omit `field`.